backend/overstroomik_service/geoserver.py

```diff
@@ -0,0 +1,152 @@
+        :param rd_y: y coordinate in meters, geocoded location latitude transformed into EPSG:28992
+        :param geoserver_url: link to the geoserver (example: http://geoserver:8080/geoserver)
+        :param layers: group layer with the expected data  (example: overstroomik:Overstroomik_data)
+        """
+
+        # initial no error
+        status = ERROR_GENERAL_NOER
+
+        # start with empty object
+        data_item = {}
+
+        # create the getfeature info url
+        api_info = self.get_api_url_from_rd(
+            rd_x=rd_x, rd_y=rd_y, geoserver_url=geoserver_url)
+
+        url = api_info.get("api_url")
+
+        # Check input, is location between the layer exent
+        if api_info.get("valid", True) is False:
+            status = ERROR_GEOS_NO_SMAP
+        else:
+            # connect async to the geoserver
+            async with httpx.AsyncClient() as client:
+
+                # fetch the feature info
+                result = await client.get(url, timeout=10.0)
+
+                if result.status_code == httpx.codes.OK:
+
+                    out = result.json()
+                    features = out.get("features")
+
+                    if len(features) > 0:
+
+                        for feature in features:
+
+                            # id or name of the sub layer
+                            f_id = feature.get("id")
+                            properties = feature.get("properties")
+
+                            if len(f_id) == 0 or f_id.startswith(
+                                    settings.LAYER_MAXIMUM_WATER_DEPTH):
+                                data_item["maximum_water_depth"] = properties.get(
+                                    settings.FIELD_MAXIMUM_WATER_DEPTH, None)
+
+                            elif f_id.startswith(settings.LAYER_SAFETY_BOARD_ID):
+                                data_item["safety_board_id"] = properties.get(
+                                    settings.FIELD_SAFETY_BOARD_ID, None)
+
+                            elif f_id.startswith(
+                                    settings.LAYER_PROBABILITY_OF_FLOODING):
+                                data_item["probability_of_flooding"] = properties.get(
+                                    settings.FIELD_PROBABILITY_OF_FLOODING, None)
+
+                            elif f_id.startswith(settings.LAYER_EVACUATION_PERCENTAGE):
+                                data_item["evacuation_percentage"] = properties.get(
+                                    settings.FIELD_EVACUATION_PERCENTAGE, None)
+
+                            elif f_id.startswith(settings.LAYER_FLOOD_TYPE):
+                                data_item["flood_type"] = properties.get(
+                                    settings.FIELD_FLOOD_TYPE, None)
+
+                        else:
+                            status: ERROR_GEOS_NO_SMAP
+                else:
+                    status = ERROR_GEOS_NO_RESP
+
+        return status, Data(**data_item)
```

Comment:
Can Data be an empty object (ie, fields can be all None, when the dict we pass is empty?)